release-notes.txt

```diff
@@ -1,3 +1,651 @@
++ dist/tools/openocd: add RTT port variable (#19723)
++ tools/build_system_sanity_check: add check to verify test
+  applications are at the right place (#19673)
+* dist/testbed-support: remove obsolete boards from iotlab archi (#19514)
+* dist/tools/insufficient_memory: fix collection of app folders (#19598)
+* dist/tools/openocd: fix parsing of flash bank base (#19620)
+* dist/tools/openocd: start debug-server in background and wait (#19737)
+* dist/tools/usb-serial: call ttys.py with its path [backport 2023.07] (#19831)
+* dist/tools: Use `/usr/bin/env` to lookup Python. (#19448)
+* make/usb_board_reset: define {preflash,term}-delay when necessary (#19617)
+* Makefile.include: don't use target lstfile implicitly (#19797)
+* Makefile.include: Generate lst file using objdump (#19745)
+* tests/drivers: move all driver tests into own folder (#19435)
+* tests: move all bench applications to their own tests/bench/ folder (#19564)
+* tests: move build system applications to tests/build_system (#19567)
+* tests: move core related applications to their own tests/core/ folder (#19565)
+* tests: move cpu related applications to tests/cpu (#19583)
+* tests: move net related applications to test/net (#19586)
+* tests: move nimble and lwip related to test applications to tests/pkg (#19582)
+* tests: move sys related applications to their own tests/sys/ folder (#19566)
+
+Kconfig
+-------
+
++ pkg/libbase58: add kconfig support (#19623)
++ sys: add Kconfig support for clif (#19621)
++ sys: add some missing kconfig + adapt related tests when possible (#19622)
+* boards/esp32s3-pros3: Fix stdio kconfig model (#19708)
+* cpu/esp32: deduplication in Kconfig for ESP32x SoCs (#19433)
+* cpu/rpx0xx: Fix kconfig model (#19660)
+* drivers/nrf24l01p: model in kconfig (#19650)
+* periph/timer: fix Kconfig menu title (#17830)
+* pkg/micropython: model in Kconfig (#19672)
+* pkg/openthread: Fix Kconfig and broken example (#19700)
+* pkg/openthread: model in kconfig (#19671)
+* Remodel the USB in Kconfig (#19086)
+* sys/stdio_semihosting: model in Kconfig (#19651)
+* sys/usb/Kconfig: Fix default PID (#19701)
+* sys: model ecc, evtimer, pipe and shell_lock in kconfig (#19636)
+
+Examples
+--------
+
+* examples/suit_update: some test fixes (#19643)
+
+Testing
+-------
+
++ tests/net/gnrc_mac_timeout: add automated test (#19639)
++ tests/README.md: Add directory overview (#19587)
+* tests/core: remove Makefile.sys_common copy past mistake (#19783)
+* tests/drivers_at24cxxx: make it easier to select other EEPROMS (#19457)
+* tests/lua_loader: move to tests/pkg/lua_loader (#19569)
+* tests/periph_rtt: Fix for tick conversion test (#19522)
+* tests/pkg/lvgl: avoid using floats (#19698)
+* tests/unittests: remove old workaround for SAML1X and gcc9.X bug (#19687)
+* tests: move leftover sys related tests to test/sys + move
+  tests/sys/candev to tests/drivers (#19585)
+* tests: move remaining driver related applications to tests/drivers (#19568)
+
+
+Uncategorized
```

Comment:
If there are uncategorized itmes, it means some labels were missing. This section should not show up in the notes.